packages/vite/src/node/build.ts

```diff
@@ -283,13 +283,50 @@ let parallelCallCounts = 0
 // bundle is even pushed.
 const parallelBuilds: RollupBuild[] = []
 
+/**
+ * Watch for file changes with chokidar
+ * Build production bundle on changes
+ */
+export async function watch(inlineConfig: InlineConfig = {}): Promise<void> {
+  const config = await resolveConfig(inlineConfig, 'build', 'production')
+  const { ignored = [], ...watchOptions } = config.build.watch || {}
+  const watcher = chokidar.watch(path.resolve(config.root), {
+    ignored: [
+      '**/node_modules/**',
+      '**/.git/**',
+      `${path.resolve(config.root, config.build.outDir)}/**`,
+      ...ignored
+    ],
+    ignoreInitial: true,
+    ignorePermissionErrors: true,
+    ...watchOptions
+  })
+
+  // do initial build before watching for changes
+  config.logger.info(chalk.cyanBright(`\nwatch mode enabled\n`))
+  await build(inlineConfig)
+  config.logger.info(chalk.cyanBright(`\nwatching for file changes...`))
+
+  watcher.on('all', async () => {
+    if (parallelCallCounts > 0) {
+      // we can not cancel current build
+      // so we need to wait until it is finished
+      return
```

Comment:
Shouldn't we trigger another build in this case once the current build is finished to avoid missing an update? In case there are several updates during a build, we should batch them in a single build once we have a chance to re-build.

Reply:
done in 03aea0c6